tide-router/src/router.rs

```diff
@@ -4,29 +4,29 @@ use futures::prelude::*;
 use http_service::Body;
 use route_recognizer::{Match, Params, Router as MethodRouter};
 
-use crate::{
-    endpoint::{DynEndpoint, Endpoint},
-    Context, Response,
-};
+use tide_core::{internal::DynEndpoint, Context, Endpoint, Response};
 
 /// The routing table used by `App`
 ///
 /// Internally, we have a separate state machine per http method; indexing
 /// by the method first allows the table itself to be more efficient.
 #[allow(missing_debug_implementations)]
-pub(crate) struct Router<State> {
+#[derive(Default)]
+pub struct Router<State> {
     method_map: FnvHashMap<http::Method, MethodRouter<Box<DynEndpoint<State>>>>,
 }
 
+// TODO (PR258): We don't want this exposed outside of tide repo
 /// The result of routing a URL
-pub(crate) struct Selection<'a, State> {
-    pub(crate) endpoint: &'a DynEndpoint<State>,
-    pub(crate) params: Params,
+#[allow(missing_debug_implementations)]
+pub struct Selection<'a, State> {
```

Comment:
We might need to merge `Context`-building logic of `tide::App` into `Router::route` in order to remove this.